lib/CModHandler.h

```diff
@@ -276,11 +276,32 @@ class DLL_LINKAGE CModHandler
 	void loadOneMod(std::string modName, std::string parent, const JsonNode & modSettings, bool enableMods);
 public:
 	
-	class Incompatibility: public std::logic_error
+	class DLL_LINKAGE Incompatibility: public std::exception
 	{
 	public:
-		Incompatibility(const std::string & w): std::logic_error(w)
-		{}
+		using StringPair = std::pair<const std::string, const std::string>;
+		using ModList = std::list<StringPair>;
+		
+		Incompatibility(ModList && _missingMods):
+			missingMods(std::move(_missingMods))
+		{
+			std::ostringstream _ss;
+			for(auto & m : missingMods)
+				_ss << m.first << ' ' << m.second << std::endl;

```

Comment:
```suggestion
			for(const auto & m : missingMods)
				_ss << m.first << ' ' << m.second << '\n';

```

Reply:
I don't think it's differs somehow for string stream but you need to be cautious with suggested approach when you are working with more than one thread

Reply:
the caution would probably apply when string stream can be shared between threads? Here it's a local variable. Or maybe I simply don't know something.

Reply:
local for sure